pkgs/applications/misc/obsidian/default.nix

```diff
@@ -29,8 +29,8 @@ let
   };
 
   icon = fetchurl {
-    url = "https://obsidian.md/images/obsidian-logo-gradient.svg";
-    hash = "sha256-EZsBuWyZ9zYJh0LDKfRAMTtnY70q6iLK/ggXlplDEoA=";
+    url = "https://forum.obsidian.md/uploads/default/original/3X/9/f/9f1b5b46aed533f5386cf276ab2cdce48cbd2e25.png";
+    hash = "sha256-GujZZXexbv7aYCzZ7uRvX1uhOQ7vU9+SaGc4ht0qzhI=";
```

Comment:
The original URL provides the new application logo and the hash is also correct.
```suggestion
    url = "https://obsidian.md/images/obsidian-logo-gradient.svg";
    hash = "sha256-EZsBuWyZ9zYJh0LDKfRAMTtnY70q6iLK/ggXlplDEoA=";
```

You can validate the Obisidian build actually returns the correct icon by running nix-build -A obisidan and then go into the directory called `result/share/icons/hicolor/128x128/apps/obsidian.png`

Reply:
It's the new logo, but it's lacking the dark background. I debated finding a version that had the BG.. but I couldn't find one that seemed "official" - at least not as official as the svg I used.

I am not really opposed to using the one with the BG if it looks good for the various resolutions.

Reply:
I think it would be better to use imagemagic to add the background instead, so we keep a good and solid reference to their logo.

Reply:
I am not sure how one would do that (with ease) the bg isn't just a solid color - it has rounded borders and gradient.. stuff..

Reply:
For now, I've made a quick hard-coded example that creates the image from the original source. It might need some small adjustments here and there to make it perfect (:

### Code
```
# Create the shadowed image
convert input.png \( +clone -background "rgba(0,0,0,0.5)" -shadow 50x5+5+5 \) +swap -background none -layers merge +repage shadowed_image.png

# Create the rounded mask
convert -size 150x150 xc:none -fill "#232129" -draw "roundrectangle 0,0,150,150,35,35" mask.png 

# Overlay the rounded image with the shadow onto the colored background
convert -gravity center mask.png shadowed_image.png -composite -compose SrcIn rounded.png
```

### Flow

#### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Input Image &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Output Image
![input](https://github.com/NixOS/nixpkgs/assets/34864484/c3aa3840-63f2-431a-82ac-dc9f6fe2926d) &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ![rounded](https://github.com/NixOS/nixpkgs/assets/34864484/c30b2b3c-3d11-4885-8796-ac67a8a7e270)

Reply:
> It's the new logo, but it's lacking the dark background. I debated finding a version that had the BG.. but I couldn't find one that seemed "official" - at least not as official as the svg I used.
> 
> I am not really opposed to using the one with the BG if it looks good for the various resolutions.

What would you think about using the icon from the Flathub repo

https://github.com/flathub/md.obsidian.Obsidian/blob/39967372569993c60d69aeedfb027bc1b56d9586/md.obsidian.Obsidian.png?raw=true

Reply:
Thanks for being patient. Again, great idea, but to quote the [README.md](https://github.com/flathub/md.obsidian.Obsidian/blob/master/README.md?plain=1#L4C8-L4C8)

> This distribution is not officially supported by the Obsidian team.

Maybe I'm being a bit picky, but I think the icon should come from an official source AND from a stable link.

Reply:
Also, to be noted from inspecting the website. Using the .svg link and then adding the background is the same method as they use on their site.